erpnext/controllers/buying_controller.py

```diff
@@ -44,6 +46,14 @@ def onload(self):
 			if self.doctype in ("Supplier Quotation", "Purchase Order", "Purchase Receipt", "Purchase Invoice"):
 				self.calculate_taxes_and_totals()
 
+		for item in self.get("items"):
+				item.update(get_bin_details(item.item_code, item.warehouse))
+				if item.meta.has_field('actual_batch_qty'):
+					if item.get('batch_no'):
+						item.actual_batch_qty = get_batch_qty(item.batch_no, item.warehouse, item.item_code)
+					else:
+						item.actual_batch_qty = 0
+
```

Comment:
- fix indentation
- can be commonified between Selling Controller and Material Request (optional)

Reply:
- remove from Material Request, Selling Controller, Buying Controller
- move to Transaction Controller
- add additional condition `if item.meta.has_field('actual_qty')` for `get_bin_details`